### DISCUSSION TITLE
Convert Superpixel to Mesh data

### DISCUSSION THREAD

**Discussion by hkim716** [pyg-team/pytorch_geometric#3969] [Q&A] [answered]
I'm trying to convert ```MNIST superpixel``` dataset to ```Mesh``` data, which has ```pos``` and ```faces```. 
I know that converting graph to mesh is now an easy task, but is there a way (or previous experience) that I could try?
Thanks!

**Comment by rusty1s** [comment, marked as answer]
You could make use of our `Delaunay` transformation.

**Reply by hkim716** [reply]
Thanks Matt. It worked!

Is there a way to assign both ```edge_attr``` and ```face``` to the data object?
When I printed


``` python
transform_graph = T.Cartesian(cat=False)
transform_mesh = T.Delaunay()

mnist_dataset_graph = MNISTSuperpixels(path, True, transform=transform_graph)[:10] # True means it's train dataset
mnist_dataset_mesh = MNISTSuperpixels(path, True, transform=transform_mesh)[:10] # True means it's train dataset

print(mnist_dataset_graph[0])
print(mnist_dataset_mesh[0])
```

I saw 
``` python
Data(edge_attr=[1399, 2], edge_index=[2, 1399], pos=[75, 2], x=[75, 1], y=[1])
Data(edge_index=[2, 1399], face=[3, 138], pos=[75, 2], x=[75, 1], y=[1])
```

How could I get datasets 
```Data(edge_attr=[1399, 2], edge_index=[2, 1399], face=[3, 138], pos=[75, 2], x=[75, 1], y=[1])``` like this efficiently?

**Reply by rusty1s** [reply]
You can compose transformations via `T.Compose`:
```python
transform = T.Compose([T.Delaunay(), T.Cartesian(cat=False)])
```

**Reply by hkim716** [reply]
When I use ```Delaunay``` transformation, then it generates new triangular meshes which has faces. 
However, it looks ```edge_index [2, 1399]``` is still edges for original graphs. 
How could I get and transform ```edge_index``` based on the newly generated triangular meshes?

**Reply by rusty1s** [reply]
You can call the `FaceToEdge` transform afterwards :)
